APCAppCore/APCAppCore/DataSubstrate/Model/APCUser+Bridge.m

```diff
@@ -81,6 +92,44 @@ - (void)signUpWithDataGroups:(NSArray<NSString *> *)dataGroups onCompletion:(voi
     }
 }
 
+- (void) signUpWithDataGroups:(NSArray<NSString *> *)dataGroups
+         withTestUserPromptVc:(__weak UIViewController*)vc
+                 onCompletion:(void (^)(NSError *))completionBlock
+{
+    [self signUpWithDataGroups:dataGroups
+          withTestUserPromptVc:vc
+              includeTestCheck:shouldPerformTestUserEmailCheckOnSignup
+                verifiedTester:NO
+                  onCompletion:completionBlock];
+}
+
+- (void) signUpWithDataGroups:(NSArray<NSString *> *)dataGroups
+         withTestUserPromptVc:(__weak UIViewController*)vc
+             includeTestCheck:(BOOL)includeTestCheck
+               verifiedTester:(BOOL)userVerifiedAsTester
+                 onCompletion:(void (^)(NSError *))completionBlock
+{
+    if (!includeTestCheck ||
+        ![[self.email lowercaseString] containsString:kHiddenTestEmailString])
+    {
+        [self signUpWithDataGroups:dataGroups onCompletion:completionBlock];
+        return;
+    }
+    
+    if (!userVerifiedAsTester)
+    {
+        [self showTestUserVerificationAlertWithDataGroups:dataGroups
+                                     withTestUserPromptVc:vc
+                                             onCompletion:completionBlock];
```

Comment:
So, if instead of just passing completionBlock through here, you pass a completion block like this:
`(BOOL testing)^{
    if (testing) {
        NSMutableArray* dataGroupsWithTest = [dataGroups mutableCopy];
        [dataGroupsWithTest addObject:kTestDataGroup];
        [self signUpWithDataGroups:dataGroupsWithTest onCompletion:completionBlock];
    } else {
        [self signUpWithDataGroups:dataGroups onCompletion:completionBlock];
    }
}`

and then the showTestUserVerificationAlert... method calls its completion block with the user's YES or NO answer, then this code doesn't need to be in a separate method with extra parameters, and I think it's a bit cleaner. No?